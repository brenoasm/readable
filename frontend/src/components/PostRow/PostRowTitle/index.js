import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import EditIcon from '../../icons/EditIcon';
import DeleteIcon from '../../icons/DeleteIcon';

const propTypes = {
  post: PropTypes.object,
  editPost: PropTypes.func,
  deletePost: PropTypes.func
};

const defaultProps = {
  post: {
    title: '',
    body: '',
    commentCount: 0,
    id: ''
  },
  editPost: () => {},
  deletePost: () => {}
};

const StyledPostRowTitle = styled.div`
  i {
    font-size: small;
  }

  a {
    cursor: pointer;
    font-weight: 600;
    font-size: 24px;
  }

  div {
    display: flex;

    > div {
      margin-left: auto;

      span {
        margin-left: 10px;
      }
    }
  }
`;

const PostRowTitle = ({ post, editPost, deletePost }) => (
  <StyledPostRowTitle>
    <Link to={`/posts/${post.id}`}>
      {post.title}
      &nbsp;
      <i title="Clique para ver a postagem" className="fas fa-link" />
    </Link>
    <p>{post.body}</p>
    <div>
      <span>
        <i title="Quantidade de comentÃ¡rios" className="far fa-comment" />
        &nbsp;
        <small>{post.commentCount}</small>
      </span>
      <div>
        {editPost && (
          <EditIcon
            handleEdit={editPost}
            objectToEdit={post}
            title="Editar esse post"
          />
        )}
        {deletePost && (
          <DeleteIcon
            handleDelete={deletePost}
            objectToDelete={post}
            title="Remover esse post"
          />
        )}
      </div>
    </div>
  </StyledPostRowTitle>
);

PostRowTitle.propTypes = propTypes;
PostRowTitle.defaultProps = defaultProps;

export default PostRowTitle;
