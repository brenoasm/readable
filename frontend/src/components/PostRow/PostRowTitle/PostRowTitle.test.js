/* eslint-disable */

import React from 'react';

import PostRowTitle from './';

describe('PostRowTitle', () => {
  it('matches the snapshot', () => {
    const initialProps = {
      post: {
        id: '6ni6ok3ym7mf1p33lnez',
        timestamp: 1468479767190,
        title: 'Learn Redux in 10 minutes!',
        body:
          'Just kidding. It takes more than 10 minutes to learn technology.',
        author: 'thingone',
        category: 'redux',
        voteScore: -5,
        deleted: false,
        commentCount: 0
      },
      editPost: jest.fn(),
      deletePost: jest.fn()
    };

    const wrapper = shallow(<PostRowTitle {...initialProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
