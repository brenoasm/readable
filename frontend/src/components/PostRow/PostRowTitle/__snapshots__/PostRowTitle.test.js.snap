// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostRowTitle matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PostRowTitle
    deletePost={[Function]}
    editPost={[Function]}
    post={
        Object {
            "author": "thingone",
            "body": "Just kidding. It takes more than 10 minutes to learn technology.",
            "category": "redux",
            "commentCount": 0,
            "deleted": false,
            "id": "6ni6ok3ym7mf1p33lnez",
            "timestamp": 1468479767190,
            "title": "Learn Redux in 10 minutes!",
            "voteScore": -5,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Link
          replace={false}
          to="/posts/6ni6ok3ym7mf1p33lnez"
>
          Learn Redux in 10 minutes!
           
          <i
                    className="fas fa-link"
                    title="Clique para ver a postagem"
          />
</Link>,
        <p>
          Just kidding. It takes more than 10 minutes to learn technology.
</p>,
        <div>
          <span>
                    <i
                              className="far fa-comment"
                              title="Quantidade de comentários"
                    />
                     
                    <small />
          </span>
          <div>
                    <EditIcon
                              handleEdit={[Function]}
                              objectToEdit={
                                        Object {
                                                  "author": "thingone",
                                                  "body": "Just kidding. It takes more than 10 minutes to learn technology.",
                                                  "category": "redux",
                                                  "commentCount": 0,
                                                  "deleted": false,
                                                  "id": "6ni6ok3ym7mf1p33lnez",
                                                  "timestamp": 1468479767190,
                                                  "title": "Learn Redux in 10 minutes!",
                                                  "voteScore": -5,
                                                }
                              }
                              title="Editar esse post"
                    />
                    <DeleteIcon
                              handleDelete={[Function]}
                              objectToDelete={
                                        Object {
                                                  "author": "thingone",
                                                  "body": "Just kidding. It takes more than 10 minutes to learn technology.",
                                                  "category": "redux",
                                                  "commentCount": 0,
                                                  "deleted": false,
                                                  "id": "6ni6ok3ym7mf1p33lnez",
                                                  "timestamp": 1468479767190,
                                                  "title": "Learn Redux in 10 minutes!",
                                                  "voteScore": -5,
                                                }
                              }
                              title="Remover esse post"
                    />
          </div>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            "Learn Redux in 10 minutes!",
            " ",
            <i
              className="fas fa-link"
              title="Clique para ver a postagem"
/>,
          ],
          "replace": false,
          "to": "/posts/6ni6ok3ym7mf1p33lnez",
        },
        "ref": null,
        "rendered": Array [
          "Learn Redux in 10 minutes!",
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-link",
              "title": "Clique para ver a postagem",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Just kidding. It takes more than 10 minutes to learn technology.",
        },
        "ref": null,
        "rendered": "Just kidding. It takes more than 10 minutes to learn technology.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              <i
                            className="far fa-comment"
                            title="Quantidade de comentários"
              />
               
              <small />
</span>,
            <div>
              <EditIcon
                            handleEdit={[Function]}
                            objectToEdit={
                                          Object {
                                                        "author": "thingone",
                                                        "body": "Just kidding. It takes more than 10 minutes to learn technology.",
                                                        "category": "redux",
                                                        "commentCount": 0,
                                                        "deleted": false,
                                                        "id": "6ni6ok3ym7mf1p33lnez",
                                                        "timestamp": 1468479767190,
                                                        "title": "Learn Redux in 10 minutes!",
                                                        "voteScore": -5,
                                                      }
                            }
                            title="Editar esse post"
              />
              <DeleteIcon
                            handleDelete={[Function]}
                            objectToDelete={
                                          Object {
                                                        "author": "thingone",
                                                        "body": "Just kidding. It takes more than 10 minutes to learn technology.",
                                                        "category": "redux",
                                                        "commentCount": 0,
                                                        "deleted": false,
                                                        "id": "6ni6ok3ym7mf1p33lnez",
                                                        "timestamp": 1468479767190,
                                                        "title": "Learn Redux in 10 minutes!",
                                                        "voteScore": -5,
                                                      }
                            }
                            title="Remover esse post"
              />
</div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="far fa-comment"
                  title="Quantidade de comentários"
/>,
                " ",
                <small />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "far fa-comment",
                  "title": "Quantidade de comentários",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 0,
                },
                "ref": null,
                "rendered": 0,
                "type": "small",
              },
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <EditIcon
                  handleEdit={[Function]}
                  objectToEdit={
                                    Object {
                                                      "author": "thingone",
                                                      "body": "Just kidding. It takes more than 10 minutes to learn technology.",
                                                      "category": "redux",
                                                      "commentCount": 0,
                                                      "deleted": false,
                                                      "id": "6ni6ok3ym7mf1p33lnez",
                                                      "timestamp": 1468479767190,
                                                      "title": "Learn Redux in 10 minutes!",
                                                      "voteScore": -5,
                                                    }
                  }
                  title="Editar esse post"
/>,
                <DeleteIcon
                  handleDelete={[Function]}
                  objectToDelete={
                                    Object {
                                                      "author": "thingone",
                                                      "body": "Just kidding. It takes more than 10 minutes to learn technology.",
                                                      "category": "redux",
                                                      "commentCount": 0,
                                                      "deleted": false,
                                                      "id": "6ni6ok3ym7mf1p33lnez",
                                                      "timestamp": 1468479767190,
                                                      "title": "Learn Redux in 10 minutes!",
                                                      "voteScore": -5,
                                                    }
                  }
                  title="Remover esse post"
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "handleEdit": [Function],
                  "objectToEdit": Object {
                    "author": "thingone",
                    "body": "Just kidding. It takes more than 10 minutes to learn technology.",
                    "category": "redux",
                    "commentCount": 0,
                    "deleted": false,
                    "id": "6ni6ok3ym7mf1p33lnez",
                    "timestamp": 1468479767190,
                    "title": "Learn Redux in 10 minutes!",
                    "voteScore": -5,
                  },
                  "title": "Editar esse post",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "handleDelete": [Function],
                  "objectToDelete": Object {
                    "author": "thingone",
                    "body": "Just kidding. It takes more than 10 minutes to learn technology.",
                    "category": "redux",
                    "commentCount": 0,
                    "deleted": false,
                    "id": "6ni6ok3ym7mf1p33lnez",
                    "timestamp": 1468479767190,
                    "title": "Learn Redux in 10 minutes!",
                    "voteScore": -5,
                  },
                  "title": "Remover esse post",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Link
            replace={false}
            to="/posts/6ni6ok3ym7mf1p33lnez"
>
            Learn Redux in 10 minutes!
             
            <i
                        className="fas fa-link"
                        title="Clique para ver a postagem"
            />
</Link>,
          <p>
            Just kidding. It takes more than 10 minutes to learn technology.
</p>,
          <div>
            <span>
                        <i
                                    className="far fa-comment"
                                    title="Quantidade de comentários"
                        />
                         
                        <small />
            </span>
            <div>
                        <EditIcon
                                    handleEdit={[Function]}
                                    objectToEdit={
                                                Object {
                                                            "author": "thingone",
                                                            "body": "Just kidding. It takes more than 10 minutes to learn technology.",
                                                            "category": "redux",
                                                            "commentCount": 0,
                                                            "deleted": false,
                                                            "id": "6ni6ok3ym7mf1p33lnez",
                                                            "timestamp": 1468479767190,
                                                            "title": "Learn Redux in 10 minutes!",
                                                            "voteScore": -5,
                                                          }
                                    }
                                    title="Editar esse post"
                        />
                        <DeleteIcon
                                    handleDelete={[Function]}
                                    objectToDelete={
                                                Object {
                                                            "author": "thingone",
                                                            "body": "Just kidding. It takes more than 10 minutes to learn technology.",
                                                            "category": "redux",
                                                            "commentCount": 0,
                                                            "deleted": false,
                                                            "id": "6ni6ok3ym7mf1p33lnez",
                                                            "timestamp": 1468479767190,
                                                            "title": "Learn Redux in 10 minutes!",
                                                            "voteScore": -5,
                                                          }
                                    }
                                    title="Remover esse post"
                        />
            </div>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "Learn Redux in 10 minutes!",
              " ",
              <i
                className="fas fa-link"
                title="Clique para ver a postagem"
/>,
            ],
            "replace": false,
            "to": "/posts/6ni6ok3ym7mf1p33lnez",
          },
          "ref": null,
          "rendered": Array [
            "Learn Redux in 10 minutes!",
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-link",
                "title": "Clique para ver a postagem",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Just kidding. It takes more than 10 minutes to learn technology.",
          },
          "ref": null,
          "rendered": "Just kidding. It takes more than 10 minutes to learn technology.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                <i
                                className="far fa-comment"
                                title="Quantidade de comentários"
                />
                 
                <small />
</span>,
              <div>
                <EditIcon
                                handleEdit={[Function]}
                                objectToEdit={
                                                Object {
                                                                "author": "thingone",
                                                                "body": "Just kidding. It takes more than 10 minutes to learn technology.",
                                                                "category": "redux",
                                                                "commentCount": 0,
                                                                "deleted": false,
                                                                "id": "6ni6ok3ym7mf1p33lnez",
                                                                "timestamp": 1468479767190,
                                                                "title": "Learn Redux in 10 minutes!",
                                                                "voteScore": -5,
                                                              }
                                }
                                title="Editar esse post"
                />
                <DeleteIcon
                                handleDelete={[Function]}
                                objectToDelete={
                                                Object {
                                                                "author": "thingone",
                                                                "body": "Just kidding. It takes more than 10 minutes to learn technology.",
                                                                "category": "redux",
                                                                "commentCount": 0,
                                                                "deleted": false,
                                                                "id": "6ni6ok3ym7mf1p33lnez",
                                                                "timestamp": 1468479767190,
                                                                "title": "Learn Redux in 10 minutes!",
                                                                "voteScore": -5,
                                                              }
                                }
                                title="Remover esse post"
                />
</div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    className="far fa-comment"
                    title="Quantidade de comentários"
/>,
                  " ",
                  <small />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "far fa-comment",
                    "title": "Quantidade de comentários",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 0,
                  },
                  "ref": null,
                  "rendered": 0,
                  "type": "small",
                },
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <EditIcon
                    handleEdit={[Function]}
                    objectToEdit={
                                        Object {
                                                            "author": "thingone",
                                                            "body": "Just kidding. It takes more than 10 minutes to learn technology.",
                                                            "category": "redux",
                                                            "commentCount": 0,
                                                            "deleted": false,
                                                            "id": "6ni6ok3ym7mf1p33lnez",
                                                            "timestamp": 1468479767190,
                                                            "title": "Learn Redux in 10 minutes!",
                                                            "voteScore": -5,
                                                          }
                    }
                    title="Editar esse post"
/>,
                  <DeleteIcon
                    handleDelete={[Function]}
                    objectToDelete={
                                        Object {
                                                            "author": "thingone",
                                                            "body": "Just kidding. It takes more than 10 minutes to learn technology.",
                                                            "category": "redux",
                                                            "commentCount": 0,
                                                            "deleted": false,
                                                            "id": "6ni6ok3ym7mf1p33lnez",
                                                            "timestamp": 1468479767190,
                                                            "title": "Learn Redux in 10 minutes!",
                                                            "voteScore": -5,
                                                          }
                    }
                    title="Remover esse post"
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "handleEdit": [Function],
                    "objectToEdit": Object {
                      "author": "thingone",
                      "body": "Just kidding. It takes more than 10 minutes to learn technology.",
                      "category": "redux",
                      "commentCount": 0,
                      "deleted": false,
                      "id": "6ni6ok3ym7mf1p33lnez",
                      "timestamp": 1468479767190,
                      "title": "Learn Redux in 10 minutes!",
                      "voteScore": -5,
                    },
                    "title": "Editar esse post",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "handleDelete": [Function],
                    "objectToDelete": Object {
                      "author": "thingone",
                      "body": "Just kidding. It takes more than 10 minutes to learn technology.",
                      "category": "redux",
                      "commentCount": 0,
                      "deleted": false,
                      "id": "6ni6ok3ym7mf1p33lnez",
                      "timestamp": 1468479767190,
                      "title": "Learn Redux in 10 minutes!",
                      "voteScore": -5,
                    },
                    "title": "Remover esse post",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
